core/types/calls.cc

```diff
@@ -1227,9 +1235,43 @@ TypePtr AppliedType::getCallArguments(const GlobalState &gs, NameRef name) const
     return getMethodArguments(gs, klass, name, targs);
 }
 
+bool MetaType::canCallNew(const GlobalState &gs) const {
+    if (isa_type<OrType>(wrapped) || isa_type<AndType>(wrapped)) {
+        return false;
+    }
+
+    if (isa_type<ClassType>(wrapped)) {
+        auto sym = cast_type_nonnull<ClassType>(wrapped).symbol;
+        if (sym == Symbols::untyped() || sym == Symbols::bottom()) {
+            return false;
+        }
+    }
+
+    if (auto *appliedType = cast_type<AppliedType>(wrapped)) {
+        if (appliedType->klass.data(gs)->isSingletonClass(gs)) {
```

Comment:
what case is this for?

Reply:
`T.class_of(String).new`, `T.class_of(MyGeneric).new`, and `T.class_of(MyGeneric[Integer]).new` all are handled here. (Had to comment this out and re-run the test to remind myself. ;))

Reply:
Can you add this as a test case?

https://sorbet.run/#%23%20typed%3A%20strict%0A%0Aclass%20MyGeneric%0A%20%20extend%20T%3A%3AGeneric%0A%0A%20%20X%20%3D%20type_template%0Aend%0A%0AMyGeneric.new

(I think that we should allow this code, and it is currently allowed on master.)

Reply:
Added (but not yet pushed). Seems to work as expected!